examples/bin/main.ml

```diff
@@ -0,0 +1,37 @@
+let go () =
+  let s = "Hello, World!\n" in
+  (* MD5 of a string *)
+  let md5 = Digestif.MD5.digest_string s in
+  Printf.printf "md5: %s\n" (Digestif.MD5.to_hex md5);
+  (* You can check this with echo 'Hello, World!' > foo then use md5sum foo
+     (Linux) or md5 foo (MacOS) on the command line. *)
+  (* We can send multiple strings too. Here, we send our string a byte at a time. *)
+  let context = ref (Digestif.MD5.init ()) in
+    for x = 0 to String.length s - 1 do
+      context := Digestif.MD5.feed_string !context ~off:x ~len:1 s
+    done;
+  (* Get the digest from the context *)
+  let md5' = Digestif.MD5.get !context in
+  Printf.printf "md5: %s\n" (Digestif.MD5.to_hex md5')
+
+(* SHA256 of a file *)
+let from_file filename =
+  (* A file's data is naturally read in chunks. *)
+  let fh = open_in_bin filename in
+  let buf = Bytes.create 1024 in
+  let context = ref (Digestif.SHA256.init ()) in
+  let fin = ref false in
+    while not !fin do
+      let n = input fh buf 0 1024 in
+        context := Digestif.SHA256.feed_bytes !context ~off:0 ~len:n buf;
+        if n = 0 then fin := true
+    done;
```

Comment:
It will be nice to show a comparative example which does not use a reference but proposes a more functional way to digest something:
```ocaml
let rec go ctx =
  let len = input fh buf 0 1024 in
  if len = 0 then Digestif.SHA256.get ctx
  else go (Digestif.SHA256.feed_bytes ctx ~off:0 ~len buf) in
go Digestif.SHA256.empty
```